### PR TITLE
Addition of Firebase Database Misconfiguration

### PR DESCRIPTION
**Pull request by shivsahni**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [ ] Your contribution is written in an active present form for as much as possible.
- [ ] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ ] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #<insert number here>.

### CODE REVIEW

File: Document/0x05d-Testing-Data-Storage.md [hunk 1]
```diff
@@ -100,6 +100,26 @@ Secure ways to retrieve the key include:
...
 
+##### Firebase Real-time Databases
+
+Firebase is a development platform with more than 15 products, and one of them is Firebase Real-time Database. It can be leveraged by application developers to store and sync data with a NoSQL cloud-hosted database. The data is stored as JSON and is synchronized in real-time to every connected client and also remains available even when the application goes offline.
+
+###### Identifying Misconfigured Firebase Instance
+
+In Jan 2018, [Appthority Mobile Threat Team (MTT)](https://cdn2.hubspot.net/hubfs/436053/Appthority%20Q2-2018%20MTR%20Unsecured%20Firebase%20Databases.pdf "Research Paper") performed security research on insecure backend services connecting to mobile applications. They discovered a misconfiguration in Firebase, which is one of the top 10 most popular data stores which could allow attackers to retrieve all the unprotected data hosted on the cloud server. The team performed the research on 2 Million+ mobile applications and found that the around 9% of Android and almost half (47%) of iOS apps that connect to a Firebase database were vulnerable.
```
Comment (commjoen) [inline Document/0x05d-Testing-Data-Storage.md]: Can you change "research paper" into "Unsecured Firebase Databases: Exposing Sensitive Data via Thousands of Mobile Apps"?

Comment (commjoen) [reply Document/0x05d-Testing-Data-Storage.md]: can you change 9% of Android into "9% of Android apps"

File: Document/0x05d-Testing-Data-Storage.md [hunk 2]
```diff
@@ -100,6 +100,26 @@ Secure ways to retrieve the key include:
...
+
+The misconfigured firebase instance can be identified by making the following network call:
```
Comment (commjoen) [inline Document/0x05d-Testing-Data-Storage.md]: can you change firebase to Firebase?

Comment (shivsahni) [reply Document/0x05d-Testing-Data-Storage.md]: made the changes.

**Pull request merged**